core/domain/classifier_services.py

```diff
@@ -305,47 +313,68 @@ def _update_classifier_training_job(classifier_training_job_model, status):
     classifier_training_job_model.put()
 
 
-def save_classifier_training_job(algorithm_id, exp_id, exp_version,
-                                 state_name, status, training_data,
-                                 job_id="None"):
-    """Checks for the existence of the model.
-    If the model exists, it is updated using _update_classifier_training_job
-        method.
-    If the model doesn't exist, it is created using
-        _create_classifier_training_job method.
+def create_classifier_training_job(algorithm_id, interaction_id, exp_id,
+                                   exp_version, state_name, training_data,
+                                   status):
+    """Creates a ClassifierTrainingJobModel in data store.
 
     Args:
         algorithm_id: str. ID of the algorithm used to generate the model.
+        interaction_id: str. ID of the interaction to which the algorithm
+            belongs.
         exp_id: str. ID of the exploration.
         exp_version: int. The exploration version at the time
             this training job was created.
         state_name: str. The name of the state to which the classifier
             belongs.
-        status: str. The status of the training job (NEW by default).
         training_data: dict. The data used in training phase.
-        job_id: str. The optional job_id which decides to create/update
-            classifier.
+        status: str. The status of the training job (NEW by default).
 
     Returns:
         job_id: str. ID of the classifier training job.
     """
-    classifier_training_job_model = (
-        classifier_models.ClassifierTrainingJobModel.get(job_id, False))
-    if classifier_training_job_model is None:
-        classifier_training_job = classifier_domain.ClassifierTrainingJob(
-            'job_id_dummy', algorithm_id, exp_id, exp_version,
-            state_name, status, training_data)
-        classifier_training_job.validate()
-        job_id = _create_classifier_training_job(classifier_training_job)
-    else:
-        classifier_training_job = get_classifier_training_job_from_model(
-            classifier_training_job_model)
-        classifier_training_job.validate()
-        _update_classifier_training_job(classifier_training_job_model,
-                                        status)
+    classifier_training_job = classifier_domain.ClassifierTrainingJob(
```

Comment:
Let's do this:

    dummy_classifier_training_job = ...('job_id_dummy', ...)
    dummy_classifier_training_job.validate()
    job_id = classifier_models.ClassifierTrainingJobModel.create(
        algorithm_id, ...)

and get rid of the private function. That would be cleaner because it doesn't give a sort of "permanent status" to the dummy object you are using solely for validation.